src/test/java/com/bakdata/kafka/integration/StreamsCleanUpTest.java

```diff
@@ -362,18 +390,43 @@ void shouldDeleteSchemaOfIntermediateTopics(final SoftAssertions softly)
         softly.assertThat(client.getAllSubjects())
                 .contains(inputSubject, internalSubject, backingSubject, manualSubject);
 
-        this.runCleanUpWithDeletion();
+        this.runCleanUp();
```

Comment:
Through topics are not affected in this case, right? Only true internal topics (ending on -repartition e.g.)

Reply:
Exactly, that's why added separate tests for intermediate topics (the ones used in `through()`/`repartition()`)

Reply:
I was not sure about that because in the diff only two tests use `runCleanUp()`

Reply:
`shouldDeleteInternalTopics` is the test that covers the case?

Reply:
yes, `shouldDeleteInternalTopics` and `shouldDeleteSchemaOfInternalTopics` cover clean up runs without the `--delete-output` flag, i.e. make sure that internal topics are completly deleted but intermediate topics are not touched.
`shouldDeleteIntermediateTopics` and `shouldDeleteSchemaOfIntermediateTopics` ensure that intermediate topics and their schemas are deleted when the flag is set.